MQTTany 0.14.3 exception running against Python 3.10.4

Hi there,
I have had to rebuild the PI controller I am running MQTTany on, and am now getting an exception. 

It appears to be because I am rebuilding/running the latest version of Ubuntu and therefore Python is a newer version... From what I can see (from a bit of googling...), this issue probably stems from the newer version of Python, and possibly would be resolved from downgrading to 3.9. But I am wondering if there is any possibility of updating this MQTTany dependency to work with Python 3.10 as well?

Details below. 

Cheers - Glen 

Versions:
- MQTTany: 0.14.3
- Python: 3.10.4
- Enironment: Rasberry PI 3 /Ubuntu 22.04.1 LTS

```
2022-08-10 00:36:34,859 [INFO ] [core                    ] MQTTany 0.14.3 starting
2022-08-10 00:36:34,987 [INFO ] [core.gpio               ] Detected board: RASPBERRY_PI_3B
2022-08-10 00:36:35,297 [ERROR] [gpio                    ] An exception occurred while running function 'load'
2022-08-10 00:36:35,301 [ERROR] [gpio                    ] 

Traceback (most recent call last):
  File "/opt/mqttany/mqttany/modules/__init__.py", line 83, in call
    retval = func(**kwargs)
  File "/opt/mqttany/mqttany/modules/gpio/core.py", line 98, in load
    conf_options = updateConfOptions(CONF_OPTIONS)
  File "/opt/mqttany/mqttany/modules/gpio/pin/__init__.py", line 57, in updateConfOptions
    conf_options = update_dict(conf_options, digital.CONF_OPTIONS)
  File "/opt/mqttany/mqttany/common.py", line 365, in update_dict
    if not isinstance(dv, collections.MutableMapping):  # type: ignore
AttributeError: module 'collections' has no attribute 'MutableMapping'

2022-08-10 00:36:35,302 [WARN ] [core                    ] Module 'gpio' load failed
```

--- a/.github/workflows/codestyle.yml
+++ b/.github/workflows/codestyle.yml
@@ -18,7 +18,7 @@ jobs:
 
     strategy:
       matrix:
-        python-version: [3.6]
+        python-version: [3.7]
 
     steps:
     - uses: actions/checkout@v2
@@ -51,7 +51,7 @@ jobs:
 
     strategy:
       matrix:
-        python-version: [3.6]
+        python-version: [3.7]
 
     steps:
     - uses: actions/checkout@v2

--- a/.github/workflows/codestyle.yml
+++ b/.github/workflows/codestyle.yml
@@ -13,7 +13,6 @@ on:
 
 jobs:
   black:
-
     runs-on: ubuntu-latest
 
     strategy:
@@ -21,32 +20,29 @@ jobs:
         python-version: [3.7]
 
     steps:
-    - uses: actions/checkout@v2
+      - uses: actions/checkout@v2
 
-    - name: Set up Python ${{ matrix.python-version }}
-      uses: actions/setup-python@v2
-      with:
-        python-version: ${{ matrix.python-version }}
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
 
-    - name: Install dependencies
-      run: |
-        python -m pip install --upgrade pip
-        pip install black \
-                    -r requirements/mqttany.txt \
-                    -r requirements/mqtt.txt \
-                    -r requirements/i2c.txt \
-                    -r requirements/led-rpi.txt \
-                    -r requirements/led-sacn.txt
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install black \
+                      -r requirements/mqttany.txt \
+                      -r requirements/mqtt.txt \
+                      -r requirements/i2c.txt \
+                      -r requirements/led-rpi.txt \
+                      -r requirements/led-sacn.txt
 
-    - name: Checking code format with black
-      run: |
-        black --version
-        black --check --verbose --diff --color ./mqttany
+      - name: Checking code format with black
+        run: |
+          black --version
+          black --check --verbose --diff --color ./mqttany
   # /black
-
-
   pyright:
-
     runs-on: ubuntu-latest
 
     strategy:
@@ -54,33 +50,33 @@ jobs:
         python-version: [3.7]
 
     steps:
-    - uses: actions/checkout@v2
+      - uses: actions/checkout@v2
 
-    - name: Set up Python ${{ matrix.python-version }}
-      uses: actions/setup-python@v2
-      with:
-        python-version: ${{ matrix.python-version }}
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
 
-    - name: Setup node.js
-      uses: actions/setup-node@v2
-      with:
-        node-version: "12"
+      - name: Setup node.js
+        uses: actions/setup-node@v2
+        with:
+          node-version: "12"
 
-    - name: Install pyright
-      run: |
-        npm install -g pyright
+      - name: Install pyright
+        run: |
+          npm install -g pyright
 
-    - name: Install dependencies
-      run: |
-        python -m pip install --upgrade pip
-        pip install -r requirements/mqttany.txt \
-                    -r requirements/mqtt.txt \
-                    -r requirements/i2c.txt \
-                    -r requirements/led-rpi.txt \
-                    -r requirements/led-sacn.txt
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install -r requirements/mqttany.txt \
+                      -r requirements/mqtt.txt \
+                      -r requirements/i2c.txt \
+                      -r requirements/led-rpi.txt \
+                      -r requirements/led-sacn.txt
 
-    - name: Checking code with pyright
-      run: |
-        pyright --version
-        pyright --project ./mqttany
+      - name: Checking code with pyright
+        run: |
+          pyright --version
+          pyright --project ./mqttany
   # /pyright

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -2,6 +2,10 @@
 
 ## Development
 
+* **Changed**
+  * Bump minimum Python version to 3.7.
+    [[#144](https://github.com/CrazyIvan359/mqttany/pull/144)]
+
 * **Added**
   * I2C Add interrupt support to MCP230xx devices. Supports interrupts via polling device rapidly or using interrupt pin and a GPIO pin on the host device.
     [[#130](https://github.com/CrazyIvan359/mqttany/pull/130)]
@@ -17,6 +21,7 @@
     [[#131](https://github.com/CrazyIvan359/mqttany/pull/131)]
   * OneWire DS18x20 devices unable to read data due to incorrect data processing causing read operation to crash. Fixes [#136](https://github.com/CrazyIvan359/mqttany/issues/136).
     [[#137](https://github.com/CrazyIvan359/mqttany/pull137)]
+  * Compatibility fixes for newer versions of Python. Fixes [#142](https://github.com/CrazyIvan359/mqttany/issues/142).
 
 ## 0.14.3
 

--- a/mqttany.code-workspace
+++ b/mqttany.code-workspace
@@ -16,13 +16,30 @@
         "files.trimTrailingWhitespace": true,
         "files.insertFinalNewline": true,
         "editor.renderWhitespace": "all",
-        "editor.formatOnSave": true,
         "editor.rulers": [
             88
         ],
+        "[python]": {
+            "editor.formatOnSave": true,
+        },
+        "[yaml]": {
+            "editor.insertSpaces": true,
+            "editor.tabSize": 2,
+            "editor.formatOnSave": true,
+            "editor.autoIndent": "keep",
+            "editor.quickSuggestions": {
+                "other": true,
+                "comments": false,
+                "strings": true
+            }
+        },
         "[markdown]": {
             "editor.wordWrap": "on",
-            "editor.quickSuggestions": false,
+            "editor.quickSuggestions": {
+                "other": "off",
+                "comments": "off",
+                "strings": "off"
+            },
             "files.trimTrailingWhitespace": false
         }
     },

--- a/mqttany/common.py
+++ b/mqttany/common.py
@@ -39,7 +39,7 @@ __all__ = [
     "update_dict",
 ]
 
-import collections
+from collections.abc import MutableMapping, Mapping
 import re
 import signal
 import typing as t
@@ -362,9 +362,9 @@ def update_dict(
     """
     for k in u:
         dv = d.get(k, {})
-        if not isinstance(dv, collections.MutableMapping):  # type: ignore
+        if not isinstance(dv, MutableMapping):  # type: ignore
             d[k] = u[k]
-        elif isinstance(u[k], collections.Mapping):  # type: ignore
+        elif isinstance(u[k], Mapping):  # type: ignore
             d[k] = update_dict(dv, u[k])
         else:
             d[k] = u[k]

--- a/mqttany/gpio/boards/base.py
+++ b/mqttany/gpio/boards/base.py
@@ -136,7 +136,7 @@ class structPin:
                 ]
             )
 
-        if not self._lock.value:
+        if not self._lock.value:  # type: ignore
             self._lock.raw = module.encode()  # type: ignore
             if log:
                 log.debug(

--- a/mqttany/pyrightconfig.json
+++ b/mqttany/pyrightconfig.json
@@ -1,11 +1,12 @@
 {
     "stubPath": "./../typings",
-    "pythonVersion": "3.6",
+    "pythonVersion": "3.7",
     "pythonPlatform": "Linux",
     "typeCheckingMode": "strict",
     "reportImportCycles": "none",
     "reportMissingTypeStubs": "warning",
     "reportUnusedImport": "warning",
     "reportUnusedVariable": "warning",
-    "reportGeneralTypeIssues": "warning"
+    "reportGeneralTypeIssues": "warning",
+    "reportTypeCommentUsage": "none"
 }

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -1,5 +1,5 @@
 [tool.black]
-target-version = ['py36', 'py37', 'py38']
+target-version = ['py37', 'py38', 'py39', 'py310']
 exclude = '''
 /(
       \.github

--- a/pyrightconfig.json
+++ b/pyrightconfig.json
@@ -2,12 +2,13 @@
     "include": [
         "./mqttany"
     ],
-    "pythonVersion": "3.6",
+    "pythonVersion": "3.7",
     "pythonPlatform": "Linux",
     "typeCheckingMode": "strict",
     "reportImportCycles": "none",
     "reportMissingTypeStubs": "warning",
     "reportUnusedImport": "warning",
     "reportUnusedVariable": "warning",
-    "reportGeneralTypeIssues": "warning"
+    "reportGeneralTypeIssues": "warning",
+    "reportTypeCommentUsage": "none"
 }

